exec/java-exec/src/main/java/org/apache/drill/exec/store/easy/EasyEVFBatchReader.java

```diff
@@ -0,0 +1,162 @@
+    return rowWriter;
+  }
+
+  @Override
+  public boolean open(FileSchemaNegotiator negotiator) {
+    this.split = negotiator.split();
+    this.loader = negotiator.build();
+    this.rowWriter = loader.writer();
+    try {
+      this.fsStream = negotiator.fileSystem().openPossiblyCompressedStream(split.getPath());
+      this.reader = new BufferedReader(new InputStreamReader(fsStream, StandardCharsets.UTF_8));
+    } catch (IOException e) {
+      throw UserException
+        .dataReadError(e)
+        .message(String.format("Failed to open input file: %s", split.getPath()))
+        .build(logger);
+    }
+    return true;
+  }
+
+  @Override
+  public boolean next() {
+    while (!rowWriter.isFull()) {
+      if (!fileIterator.hasNext()) {
+        return false;
+      }
+      fileIterator.next();
+    }
+    return true;
+  }
+
+  @Override
+  public void close() {
+    try {
+      if (reader != null) {
+        reader.close();
+        reader = null;
+      }
+      if (fsStream != null) {
+        fsStream.close();
+        fsStream = null;
+      }
+    } catch (IOException e) {
+      logger.warn("Error closing batch Record Reader.");
+    }
+  }
+
+  /**
+   * This function should be used when writing Drill columns when the schema is not fixed or known in advance. At present only simple data types are
+   * supported with this function.  If the column is not present in the schema, this function will add it first.
+   * @param rowWriter The RowSetWriter to which the data is to be written
+   * @param name The field name
+   * @param value The field value.  Cannot be a primitive.
+   * @param type The Drill data type of the field.
+   */
+  public static void writeColumn(TupleWriter rowWriter, String name, Object value, TypeProtos.MinorType type) {
+    int index = rowWriter.tupleSchema().index(name);
+    if (index == -1) {
+      ColumnMetadata colSchema = MetadataUtils.newScalar(name, type, TypeProtos.DataMode.OPTIONAL);
+      index = rowWriter.addColumn(colSchema);
+    }
+    ScalarWriter colWriter = rowWriter.scalar(index);
+    switch (type) {
+      case INT:
+        colWriter.setInt((Integer)value);
+        break;
+      case BIGINT:
+        colWriter.setLong((Long)value);
+        break;
+      case VARCHAR:
+        colWriter.setString((String)value);
+        break;
+      case FLOAT4:
+      case FLOAT8:
+        colWriter.setDouble((Double)value);
+        break;
+      case DATE:
+        colWriter.setDate((LocalDate) value);
+        break;
+      case BIT:
+        colWriter.setBoolean((Boolean)value);
+        break;
+      case TIMESTAMP:
+      case TIMESTAMPTZ:
+        colWriter.setTimestamp((Instant) value);
+        break;
+      case TIME:
+        colWriter.setTime((LocalTime) value);
+        break;
+      case INTERVAL:
+        colWriter.setPeriod((Period)value);
+        break;
+      default:
+        logger.warn("Does not support data type {}", type.toString());
```

Comment:
We can add more details, column name for instance...

Reply:
I removed this method